kafka-client/src/main/java/dev/responsive/db/CassandraFactSchema.java

```diff
@@ -210,6 +229,7 @@ public BoundStatement delete(
       final int partitionKey,
       final Bytes key
   ) {
+    // TODO: consider throwing an exception here as TWCS doesn't work well with deletes
```

Comment:
this seems like it would be extreme -- but perhaps we could/should log a warning?

Reply:
I originally had a warn log, but didn't want to log a warning on every delete as that could quickly spam the logs. I think disabling deletes and forcing a TTL could be reasonable. This is a quote from the [docs](https://opensource.docs.scylladb.com/stable/kb/compaction.html#time-window-compactionstrategy-twcs):
> Avoid overwriting data and deleting data explicitly at all costs, as this can potentially block an expired SSTable from being purged, due to the checks that are performed to avoid data resurrection.

Reply:
Wait, it can't handle overwrites?? That's much different than just saying "avoid deletes", I'm a bit surprised/concerned to see that included in their warning 🤔 

W.r.t how we handle things on our end, do you think even a pattern of extremely rare deletes would still blow up the cluster significantly? Because if so then yeah, we should just throw an exception here. But if there's a chance of it being viable for some use cases (eg very low traffic in general with even more rare deletes) then I think logging a warning would be appropriate.
It would still be annoying enough that anyone abusing the ttl store with frequent deletes would be unable to ignore the flood of warnings blowing up their logs, but advanced users who truly understand the system and know what they're doing could still use it and just disable the Logger. Either way it makes them stop and think about whether they're doing the right thing here.

Of course this really hinges on there being any kind of valid use case -- if we believe any deletes whatsoever are essentially fatal then we can/should block it

Reply:
Like I understand _why_ this is so bad, but I'm curious _how_  bad it can get -- I guess this comes down to the size of SST tables vs frequency of deletes --> how many "dead" tables need to be kept around longer just for tombstones? I don't suppose you can get around this problem by configuring the tombstones with a very very short additional retention time (assuming that's basically how it works)

Reply:
> Wait, it can't handle overwrites

it can, but it's not the intended use case and the performance can be a struggle. clearly they found it a common enough pattern to provide it OOTB and recommend it for some use cases

> W.r.t how we handle things on our end, do you think even a pattern of extremely rare deletes would still blow up the cluster significantly?

the answer to that is probably "it depends" 😕 I have to test it to get a better sense 

>  just disable the Logger.

that was also my line of thinking, but I think there may be other valuable log statements at WARN eventually from this class. we can play it by ear then or extract this into a utility class to make it easier to disable the log

> but I'm curious how bad it can get

me too! I think there's essentially two possible outcomes: leaked disk and increased CPU. We can handle the first, the second could be fatal. It's probably worth running some kind of test.